Spec/RealtimeClientConnection.swift

```diff
@@ -2093,6 +2093,39 @@ class RealtimeClientConnection: QuickSpec {
                         expect(channel.errorReason).to(beIdenticalTo(protocolError.error))
                     }
 
+                    it("should detach all channels if the connectionId has changed") {
+                        let options = AblyTests.commonAppSetup()
+                        options.disconnectedRetryTimeout = 1.0
+                        let client = AblyTests.newRealtime(options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("test")
+
+                        expect(client.connection.state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+
+                        let expectedConnectionId = client.connection.id
+
+                        guard let transport = client.transport as? TestProxyTransport else {
+                            fail("TestProxyTransport is not set"); return
+                        }
+
+                        let disconnectedMessage = ARTProtocolMessage()
+                        disconnectedMessage.action = .Disconnected
+                        disconnectedMessage.connectionId = "xxxxxxxxx"
+                        transport.receive(disconnectedMessage)
+
+                        waitUntil(timeout: testTimeout) { done in
+                            // Resumed
+                            client.connection.once(.Connected) { stateChange in
+                                guard let stateChange = stateChange else {
+                                    fail("ConnectionStateChange is nil"); done(); return
+                                }
+                                expect(client.connection.id).to(equal(disconnectedMessage.connectionId))
```

Comment:
Why would this be equal to the fabricated message that was received?  Realtime will assign a new `connectionId` as that ID is now invalid